Library/Homebrew/dev-cmd/create.rb

```diff
@@ -140,15 +140,12 @@ def create_cask(args:)
   end
 
   def create_formula(args:)
-    fc = FormulaCreator.new(args)
-    fc.name = if args.set_name.blank?
+    fc = FormulaCreator.new(args.set_name, args.set_version, !args.no_fetch?, args.HEAD?)
+    if fc.name.blank?
       stem = Pathname.new(args.named.first).stem.rpartition("=").last
       print "Formula name [#{stem}]: "
-      __gets || stem
-    else
-      args.set_name
+      fc.name = __gets || stem
     end
-    fc.version = args.set_version
     fc.license = args.set_license
     fc.tap = Tap.fetch(args.tap || "homebrew/core")
```

Comment:
How about putting these and/or `fc.mode` in too?

Reply:
@abitrolly Yup, that should be fine; it doesn't actually install the tap. Thanks for checking!

Reply:
Moved the line to constructor. Not sure about the validation code that raises errors next line.